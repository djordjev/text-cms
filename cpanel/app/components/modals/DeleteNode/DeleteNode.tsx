import { Form, useResolvedPath, useSearchParams } from '@remix-run/react';
import { FC } from 'react';

import { Modal, ModalProps } from '~/components/modals/Modal';
import { BUTTON_ACTION } from '~/constants';

export interface DeleteNodeProps extends Omit<ModalProps, 'children'> {}

const DeleteNode: FC<DeleteNodeProps> = (props) => {
  const { onClose, open } = props;

  // Hooks
  const { pathname } = useResolvedPath('.');
  const [params] = useSearchParams();

  // Setup
  const filename = params.get('path') ?? '';

  return (
    <Modal onClose={onClose} open={open} title="Are you sure?">
      <Form
        action={pathname}
        className="u-flex u-flex-col u-modal-action"
        method="POST"
      >
        <input type="hidden" name="node" value={filename} />
        <p>
          Are you sure you want to delete{' '}
          <span className="u-font-bold u-text-primary">{filename}</span>? This
          action is irreversible and will delete all content inside of it!
        </p>

        <div className="u-divider" />

        <div className="u-grid u-grid-cols-2 u-gap-3x">
          <button
            className="u-btn u-btn-secondary u-uppercase"
            onClick={onClose}
            type={'button'}
          >
            No
          </button>

          <button
            className="u-btn u-btn-error u-uppercase"
            name={BUTTON_ACTION}
            type="submit"
            value="delete"
          >
            Yes, delete
          </button>
        </div>
      </Form>
    </Modal>
  );
};

export { DeleteNode };
