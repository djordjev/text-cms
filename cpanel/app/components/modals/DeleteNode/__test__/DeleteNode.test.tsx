import { createRemixStub } from '@remix-run/testing';
import { fireEvent, render, screen } from '@testing-library/react';

import { DeleteNode, DeleteNodeProps } from '../DeleteNode';

test('DeleteNode', () => {
  const onClose = vi.fn();

  const props: DeleteNodeProps = { onClose, open: true };

  const createComponent = () => {
    const RemixStub = createRemixStub([
      { path: '/', Component: () => <DeleteNode {...props} /> }
    ]);

    return <RemixStub initialEntries={['/?path=filename.txt']} />;
  };

  const { baseElement } = render(createComponent());

  // Renders title
  expect(screen.getByText('Are you sure?')).toBeInTheDocument();

  // Renders body
  expect(screen.getByText(/you want to delete/)).toBeInTheDocument();
  expect(screen.getByText('filename.txt')).toBeInTheDocument();

  // Renders buttons
  expect(screen.getByRole('button', { name: 'No' })).toBeInTheDocument();
  expect(
    screen.getByRole('button', { name: 'Yes, delete' })
  ).toBeInTheDocument();

  // Handles close click
  fireEvent.click(screen.getByRole('button', { name: 'close' }));
  expect(onClose).toHaveBeenCalledOnce();

  expect(baseElement).toMatchSnapshot();
});
