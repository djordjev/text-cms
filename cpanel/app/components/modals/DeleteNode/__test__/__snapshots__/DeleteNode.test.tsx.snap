// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`DeleteNode 1`] = `
<body>
  <div
    aria-hidden="true"
  >
    <div
      style="position: fixed; top: 1px; left: 1px; width: 1px; height: 0px; padding: 0px; margin: -1px; overflow: hidden; clip: rect(0px, 0px, 0px, 0px); white-space: nowrap; border-width: 0px; display: none;"
    />
  </div>
  <div
    id="headlessui-portal-root"
  >
    <div
      data-headlessui-portal=""
    >
      <button
        aria-hidden="true"
        data-headlessui-focus-guard="true"
        style="position: fixed; top: 1px; left: 1px; width: 1px; height: 0px; padding: 0px; margin: -1px; overflow: hidden; clip: rect(0px, 0px, 0px, 0px); white-space: nowrap; border-width: 0px;"
        type="button"
      />
      <div>
        <div
          aria-labelledby="headlessui-dialog-title-:r3:"
          aria-modal="true"
          class="u-modal u-modal-open"
          data-headlessui-state="open"
          id="headlessui-dialog-:r0:"
          role="dialog"
        >
          <div
            aria-hidden="true"
            class="u-fixed u-bg-black u-inset-0 u-opacity-15 u-modal-backdrop"
            data-headlessui-state="open"
            id="headlessui-dialog-overlay-:r1:"
          />
          <div
            class="u-modal-box"
            data-headlessui-state="open"
            id="headlessui-dialog-panel-:r2:"
          >
            <h2
              class="u-font-bold u-text-lg u-flex u-items-center"
              data-headlessui-state="open"
              id="headlessui-dialog-title-:r3:"
            >
              <span
                class="u-flex-1 u-uppercase u-text-primary"
              >
                Are you sure?
              </span>
              <button
                aria-label="close"
                type="button"
              >
                <svg
                  class="tabler-icon tabler-icon-x"
                  fill="none"
                  height="28"
                  stroke="currentColor"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="2"
                  viewBox="0 0 24 24"
                  width="28"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M18 6l-12 12"
                  />
                  <path
                    d="M6 6l12 12"
                  />
                </svg>
              </button>
            </h2>
            <form
              action="/"
              class="u-flex u-flex-col u-modal-action"
              method="post"
            >
              <input
                name="node"
                type="hidden"
                value="filename.txt"
              />
              <p>
                Are you sure you want to delete
                 
                <span
                  class="u-font-bold u-text-primary"
                >
                  filename.txt
                </span>
                ? This action is irreversible and will delete all content inside of it!
              </p>
              <div
                class="u-divider"
              />
              <div
                class="u-grid u-grid-cols-2 u-gap-3x"
              >
                <button
                  class="u-btn u-btn-secondary u-uppercase"
                  type="button"
                >
                  No
                </button>
                <button
                  class="u-btn u-btn-error u-uppercase"
                  name="_action"
                  type="submit"
                  value="delete"
                >
                  Yes, delete
                </button>
              </div>
            </form>
          </div>
        </div>
      </div>
      <button
        aria-hidden="true"
        data-headlessui-focus-guard="true"
        style="position: fixed; top: 1px; left: 1px; width: 1px; height: 0px; padding: 0px; margin: -1px; overflow: hidden; clip: rect(0px, 0px, 0px, 0px); white-space: nowrap; border-width: 0px;"
        type="button"
      />
    </div>
  </div>
</body>
`;
